test/com/facebook/buck/cli/testdata/output_path/output_path_json_all.js

```diff
@@ -24,7 +24,7 @@
 {
   "buck.base_path":"",
   "buck.direct_dependencies":[],
-  "buck.ruleType":"java_annotation_processor",
+  "buck.ruleType":"java_annotation_processor_plugin",
```

Comment:
What are these `.js` file changes?

Reply:
Fixing the tests as the rule type changed

Reply:
But what is `java_annotation_processor_plugin` (vs. `java_annotation_processor`/`java_plugin`)?

Reply:
The `JavaAnnotationProcessor` BuildRule name changed for `JavaAnnotationProcessorPlugin` to conform with the names (easy to know that it is a plugin after all).
This test is coupled with the name of the build rule

Reply:
All right, it will not make much sense on the perspective of code maintenance, but sure, I can change it